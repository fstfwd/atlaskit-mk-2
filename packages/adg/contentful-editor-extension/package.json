{
  "name": "ce",
  "version": "0.0.0",
  "description": "editor extension for contentful",
  "license": "Apache-2.0",
  "author": "Atlassian Pty Ltd",
  "scripts": {
    "start": "./node_modules/@atlaskit/webpack-config/bin/dev.js ./src/index.js",
    "webpack:build": "./node_modules/@atlaskit/webpack-config/bin/build.js ./src/index.js",
    "build": "WEBSITE_ENV=production yarn webpack:build"
  },
  "maintainers": [
    "Ajay M"
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "devDependencies": {
    "enzyme": "https://registry.npmjs.org/@pgleeson/enzyme/-/enzyme-3.3.7.tgz"
  },
  "keywords": [
    "atlaskit",
    "react",
    "ui"
  ],
  "dependencies": {
    "@atlaskit/avatar": "^14.0.2",
    "@atlaskit/editor-common": "^13.0.2",
    "@atlaskit/editor-confluence-transformer": "^5.0.1",
    "@atlaskit/editor-core": "^76.0.2",
    "@atlaskit/editor-json-transformer": "^4.0.1",
    "@atlaskit/editor-markdown-transformer": "^2.0.1",
    "@atlaskit/icon": "^13.1.1",
    "@atlaskit/media-core": "^22.0.0",
    "@atlaskit/webpack-config": "^2.0.2",
    "babel-loader": "^7.1.2",
    "babel-preset-env": "^1.6.0",
    "babel-preset-react": "^6.24.1",
    "express": "^4.16.3",
    "react": "^16.4.0",
    "react-loadable": "^5.1.0",
    "styled-components": "^3.2.6",
    "webpack": "^3.10.0",
    "webpack-dev-middleware": "^3.1.3",
    "webpack-dev-server": "^2.11.1"
  }
}