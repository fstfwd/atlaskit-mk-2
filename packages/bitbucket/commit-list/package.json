{
  "name": "@atlassian/bitkit-commit-list",
  "publishConfig": {
    "registry": "https://packages.atlassian.com/api/npm/atlassian-npm"
  },
  "version": "1.0.0",
  "description": "A React component for rendering a list of repository commits",
  "repository": {
    "type": "git",
    "url": "https://bitbucket.org/atlassian/atlaskit-mk-2"
  },
  "main": "lib/index.js",
  "module": "es/index.js",
  "atlaskit:src": "src/index.js",
  "atlaskit": {
    "name": "Commit List",
    "team": "bitbucket"
  },
  "files": [
    "lib",
    "es",
    "umd"
  ],
  "scripts": {
    "build": "node ../../tools/build-package.js",
    "prepublishOnly": "npm run build"
  },
  "author": "Bitbucket Cloud <bitbucket-dev@atlassian.com>",
  "maintainers": [
    "Ben G",
    "Stacy L"
  ],
  "license": "Apache-2.0",
  "dependencies": {
    "@atlaskit/avatar": "^14.0.3",
    "@atlaskit/button": "^9.0.2",
    "@atlaskit/icon": "^13.2.0",
    "@atlaskit/spinner": "^9.0.2",
    "@atlaskit/theme": "^6.0.3",
    "@atlassian/bitkit-avatars": "^5.0.0",
    "@atlassian/bitkit-builds": "^5.0.0",
    "@atlassian/bitkit-date": "^4.0.0",
    "@atlassian/bitkit-flow-types": "^0.10.0"
  },
  "peerDependencies": {
    "enzyme": "https://registry.npmjs.org/@pgleeson/enzyme/-/enzyme-3.3.7.tgz",
    "enzyme-adapter-react-16": "https://registry.npmjs.org/@pgleeson/enzyme-adapter-react-16/-/enzyme-adapter-react-16-1.1.7.tgz",
    "enzyme-react-intl": "^2.0.1",
    "enzyme-to-json": "^3.3.0",
    "react": "^16.4.0",
    "react-intl": "^2.6.0",
    "styled-components": "^3.2.6"
  },
  "devDependencies": {
    "@atlaskit/docs": "^5.0.1",
    "@atlaskit/navigation": "^33.0.2",
    "enzyme": "https://registry.npmjs.org/@pgleeson/enzyme/-/enzyme-3.3.7.tgz",
    "react-dom": "^16.4.2",
    "react-intl": "^2.6.0",
    "react-test-renderer": "^16.3.0"
  }
}