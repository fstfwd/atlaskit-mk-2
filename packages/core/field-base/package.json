{
  "name": "@atlaskit/field-base",
  "version": "9.0.0",
  "description": "Wraps fields and provides styles for focus/hover states",
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "atlaskit:src": "src/index.js",
  "files": [
    "dist"
  ],
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    {
      "name": "Mike G"
    },
    {
      "name": "Luke B"
    }
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/icon": "^11.0.0",
    "@atlaskit/inline-dialog": "^6.0.0",
    "@atlaskit/spinner": "^5.0.0",
    "@atlaskit/theme": "^3.0.0",
    "babel-runtime": "^6.26.0",
    "styled-components": "^3.2.6"
  },
  "peerDependencies": {
    "react": "^16.3.0"
  },
  "devDependencies": {
    "@atlaskit/docs": "^3.0.0",
    "@atlaskit/input": "^2.0.0",
    "enzyme": "https://registry.npmjs.org/@pgleeson/enzyme/-/enzyme-3.3.2.tgz"
  },
  "atlaskit": {
    "internal": true
  },
  "keywords": [
    "atlaskit",
    "react",
    "ui"
  ]
}