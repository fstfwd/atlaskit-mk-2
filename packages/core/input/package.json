{
  "name": "@atlaskit/input",
  "version": "3.0.2",
  "description": "Internal component to render a simple HTML input element.It is used in inline-edit and in field-base to create field-text",
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "atlaskit:src": "src/index.js",
  "files": [
    "dist"
  ],
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Sean C",
    "Charles L",
    "Mike G"
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/theme": "^4.0.4",
    "babel-runtime": "^6.26.0",
    "keycode": "^2.1.7"
  },
  "peerDependencies": {
    "react": "^16.3.0",
    "react-dom": "^16.3.0",
    "styled-components": "^3.2.6"
  },
  "devDependencies": {
    "@atlaskit/field-base": "^10.1.2",
    "@atlaskit/docs": "^4.1.1",
    "enzyme": "https://registry.npmjs.org/@pgleeson/enzyme/-/enzyme-3.3.7.tgz"
  },
  "atlaskit": {
    "internal": true
  },
  "keywords": [
    "atlaskit",
    "ui",
    "internal"
  ]
}