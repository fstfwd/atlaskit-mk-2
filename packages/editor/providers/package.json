{
  "name": "@atlaskit/providers",
  "version": "0.0.0",
  "description": "Editor Provider Package",
  "license": "Apache-2.0",
  "main": "dist/es5/index.js",
  "atlaskit:src": "src/index.ts",
  "jsnext:main": "dist/es2015/index.js",
  "module": "dist/es2015/index.js",
  "types": "dist/es5/index.d.ts",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Scott S"
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/activity": "^1.0.1",
    "@atlaskit/editor-core": "^72.1.4",
    "@atlaskit/emoji": "^35.0.3",
    "@atlaskit/mention": "^12.0.0",
    "@atlaskit/task-decision": "^6.0.7"
  },
  "peerDependencies": {
    "react": "^16.3.0"
  },
  "devDependencies": {
    "@atlaskit/docs": "^3.0.4",
    "@atlaskit/editor-common": "^10.1.0",
    "@atlaskit/media-core": "^18.1.1",
    "@types/prosemirror-model": "^1.0.0",
    "@types/prosemirror-state": "^1.0.0",
    "@types/react": "^16.0.5",
    "chai-as-promised": "^6.0.0"
  }
}