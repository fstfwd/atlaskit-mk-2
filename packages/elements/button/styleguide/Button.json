{
    "package": "@atlaskit/Button",
    "component": "Button",
    "sketchSymbol": "Button",
    "description": "Button style is defined for different appearance types, states & events",
    "htmlUrl": "/styleguides/elements/button/ButtonStyleGuide",
    "sketchUrl": "/styleguides/elements/button/ButtonStyleGuide.sketch",
    "themes": {
        "options": [
            "default",
            "dark"
        ]
    },
    "props": [
        {   
            "name": "type",
            "prop" : "appearance",
            "values" : [
                "primary",
                "default"
            ]
        },
        { "name": "selected", "prop": "isSelected", "values":[ true,false ] },
        { "name": "disabled", "prop": "isDisabled", "values":[ true,false ] },
        { "name": "width",
            "prop": "isFitContainer", 
            "values":[ 
                { "name":"full", "value": true },
                { "name":"normal", "value": false }
    ],
    "events": [
        { "name": "focus", "state": "isFocus" },
        { "name": "hover", "state": "isEvent" },
        { "name": "active", "state": "isActive" }
    ],
    "render":{
        "groups":[
            {
                "label": "All Styles", 
                "include": [ "props", "events" ],
                "textProp": "children"
            },
            {
                "label": "Full Width",
                "props": {
                    "isFitContainer": true
                }
            }
        ]
    }
}
   