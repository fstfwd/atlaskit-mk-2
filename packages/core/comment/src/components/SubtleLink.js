// @flow

import React, { type Node } from 'react';
import {
  withAnalyticsEvents,
  createAndFireEvent,
  type UIAnalyticsEvent,
} from '@atlaskit/analytics-next';
import Button from '@atlaskit/button';
import { name, version } from '../../package.json';

type Props = {
  /** The content to render inside the action button. */
  children?: Node,
  /** Handler called when the element is clicked. The second argument can be used
   * to track analytics events. See documentation in analytics-next package for details. */
  onClick?: (e: SyntheticEvent<>, analyticsEvent: UIAnalyticsEvent) => void,
  /** Handler called when the element is focused. The second argument can be used
   * to track analytics events. See documentation in analytics-next package for details. */
  onFocus?: (e: SyntheticEvent<>, analyticsEvent: UIAnalyticsEvent) => void,
  /** Handler called when the element is moused over. The second argument can be used
   * to track analytics events. See documentation in analytics-next package for details. */
  onMouseOver?: (e: SyntheticEvent<>, analyticsEvent: UIAnalyticsEvent) => void,
  /** Used to replace the button's analytics context. */
  analyticsContext: { [string]: any },
};

const SubtleLink = ({
  onClick,
  onFocus,
  onMouseOver,
  analyticsContext,
  children,
}: Props) => (
  /* eslint-disable jsx-a11y/no-static-element-interactions */
  <span onClick={onClick} onFocus={onFocus} onMouseOver={onMouseOver}>
    <Button
      appearance="subtle-link"
      spacing="none"
      type="button"
      analyticsContext={{ ...analyticsContext, package: name, version }}
    >
      {children}
    </Button>
  </span>
  /* eslint-enable jsx-a11y/no-static-element-interactions */
);

export { SubtleLink as SubtleLinkWithoutAnalytics };

const createAndFireEventOnAtlaskit = createAndFireEvent('atlaskit');
export default withAnalyticsEvents({
  onClick: createAndFireEventOnAtlaskit({ action: 'click' }),
  onFocus: createAndFireEventOnAtlaskit({ action: 'focus' }),
  onMouseOver: createAndFireEventOnAtlaskit({ action: 'mouseover' }),
})(SubtleLink);
