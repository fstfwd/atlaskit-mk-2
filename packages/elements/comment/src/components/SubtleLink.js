// @flow

import React, { type Node } from 'react';
import {
  withAnalyticsEvents,
  createAndFireEvent,
} from '@atlaskit/analytics-next';
import Button from '@atlaskit/button';

type Props = {
  /** The content to render inside the action button. */
  children?: Node,
  /** Handler called when the element is clicked. */
  onClick?: Function,
  /** Handler called when the element is focused. */
  onFocus?: Function,
  /** Handler called when the element is moused over. */
  onMouseOver?: Function,
  /** Used to replace the button's analytics context. */
  analyticsContext: { [string]: any },
};

const SubtleLink = ({
  onClick,
  onFocus,
  onMouseOver,
  analyticsContext,
  children,
}: Props) => (
  /* eslint-disable jsx-a11y/no-static-element-interactions */
  <span onClick={onClick} onFocus={onFocus} onMouseOver={onMouseOver}>
    <Button
      appearance="subtle-link"
      spacing="none"
      type="button"
      analyticsContext={analyticsContext}
    >
      {children}
    </Button>
  </span>
  /* eslint-enable jsx-a11y/no-static-element-interactions */
);

const createAndFireEventOnAtlaskit = createAndFireEvent('atlaskit');
export default withAnalyticsEvents({
  onClick: createAndFireEventOnAtlaskit({ action: 'click' }),
  onFocus: createAndFireEventOnAtlaskit({ action: 'focus' }),
  onMouseOver: createAndFireEventOnAtlaskit({ action: 'mouseover' }),
})(SubtleLink);
