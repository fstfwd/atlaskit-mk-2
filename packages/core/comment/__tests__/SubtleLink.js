// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Button from '@atlaskit/button';
import { name, version } from '../package.json';
import SubtleLink from '../src/components/SubtleLink';

describe('@atlaskit comments', () => {
  describe('SubtleLink', () => {
    it('should override analytics context of button', () => {
      const wrapper = shallow(
        <SubtleLink analyticsContext={{ component: 'comment-action' }}>
          Like
        </SubtleLink>,
      ).dive();
      const { analyticsContext } = wrapper.find(Button).props();
      expect(analyticsContext).toEqual({
        component: 'comment-action',
        package: name,
        version,
      });
    });
  });
});
