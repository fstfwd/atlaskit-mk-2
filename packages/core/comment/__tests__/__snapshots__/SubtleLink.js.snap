// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@atlaskit comments SubtleLink should render a span and a button 1`] = `
<span
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOver={[Function]}
>
  <WithAnalyticsContext
    analyticsContext={
      Object {
        "component": "comment-action",
        "package": "@atlaskit/comment",
        "version": "2.7.2",
      }
    }
    appearance="subtle-link"
    spacing="none"
    type="button"
  >
    Like
  </WithAnalyticsContext>
</span>
`;
